bulbs/content/managers.py

```diff
@@ -0,0 +1,105 @@
+"""Generic manager for all indexable content and common queries."""
+from django.conf import settings
+from django.utils import timezone
+
+from djes.models import IndexableManager
+from elasticsearch_dsl import filter as es_filter
+from polymorphic import PolymorphicManager
+
+from .filters import (
+    AllSponsored, Authors, Evergreen, FeatureTypes, Published, Status, Tags, VideohubChannel
+)
+
+
+class ContentManager(PolymorphicManager, IndexableManager):
+    """a specialized version of `djes.models.SearchManager` for `bulbs.content.Content`."""
+
+    def evergreen(self, **kwargs):
+        """
+        Search containing any evergreen piece of Content.
+
+        :included_channel_ids list: Contains ids for channel ids relevant to the query.
+        :excluded_channel_ids list: Contains ids for channel ids excluded from the query.
+        """
+        eqs = self.search(**kwargs)
+        eqs = eqs.filter(Evergreen())
+        included_channel_ids = kwargs.get("included_channel_ids", [])
+        excluded_channel_ids = kwargs.get("exncluded_channel_ids", [])
+
+        if included_channel_ids:
+            eqs = eqs.filter(VideohubChannel(included_ids=included_channel_ids))
+        if excluded_channel_ids:
+            eqs = eqs.filter(VideohubChannel(excluded_ids=excluded_channel_ids))
+        return eqs
+
+    def evergreen_video(self, **kwargs):
```

Comment:
FYI: This lacks the ability to filter out "Onion Labs" videos, which is required for Onion